src/kontrol/kdist/assert.md

```diff
@@ -75,6 +75,19 @@ Capturing cheat code calls
         orBool SELECTOR ==Int selector ( "assertEq(uint256,uint256)" )
     [preserves-definedness]
 
+    rule [cheatcode.call.assertEq.Darray]:
+         <k> #cheatcode_call SELECTOR ARGS => 
+               #let LEN_INPUT1 = #asWord(#range(ARGS, 0, 32)) #in
+               #let OFFSET_INPUT1 = 32 +Int LEN_INPUT1 *Int 32 #in
+               #let LEN_INPUT2 = #asWord(#range(ARGS, OFFSET_INPUT1, 32)) #in
+                  #assert_eq #asWord(#range(ARGS, 32, LEN_INPUT1 *Int 32)) #asWord(#range(ARGS, 32 +Int OFFSET_INPUT1, LEN_INPUT2 *Int 32)) String2Bytes("assertion failed") ... </k>
```

Comment:
Hmm, why do you need `*Int 32` in ` LEN_INPUT1 *Int 32`?

Reply:
Hi @anvacaru. I just added the encoding for the relative parts in the [first comment](https://github.com/runtimeverification/kontrol/pull/605#issue-2337730454) of this PR.

It is because `enc(T[]) = enc(k) enc(T[0] T[1] ... T[k-1]) = 32 + k * 32` where `k = len(T[])` and each `enc(T[]) = 32 bytes`, similar for `enc(k) = 32 bytes`.

Reply:
Thank you!